src/main/java/org/mitre/hapifhir/client/BearerAuthServerClient.java

```diff
@@ -0,0 +1,43 @@
+package org.mitre.hapifhir.client;
+
+import ca.uhn.fhir.rest.api.MethodOutcome;
+import ca.uhn.fhir.rest.client.api.IGenericClient;
+
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.hl7.fhir.r4.model.Bundle;
+
+public class BearerAuthServerClient implements IServerClient {
+   
+    private String token;
+    private IGenericClient client;
+
+    public BearerAuthServerClient(String token, IGenericClient client) {
```

Comment:
This is nice but it depends on a single token being valid forever, like the admin token, and things like that (hopefully) aren't used in the real world. One thing we could do to make this more generically useful is to have a second constructor take in a function to fetch a token, ie a  `java.util.function.Supplier<String>` with signature `() -> { ... return token; };`, and then the functions below would either use the single token or call the function to get one.
Or alternatively let's take this impl into the medmorph-fhir-server

Reply:
I was planning on waiting for [MEDMORPH-103](https://jira.mitre.org/browse/MEDMORPH-103) since that would implement all of those token methods. Once we have that it would be easy to re-implement this interface. I like the idea of having it pass in a Supplier but creating a new SmartBackendAuthServerClient would be nice too so others using that scheme don't need to reimplement the token methods